unblob/processing.py

```diff
@@ -326,6 +327,8 @@ def process(self):
             )
             outer_chunks = remove_inner_chunks(all_chunks)
             unknown_chunks = calculate_unknown_chunks(outer_chunks, self.size)
+            patch_chunks(outer_chunks, host_file=self.task.path)
+            patch_chunks(unknown_chunks, host_file=self.task.path)
```

Comment:
just wondering why can not we do this "patching" in `_hyperscan_match` after the `_calculate_chunk` call?

We have the file and also the file size there available, we can maybe even use it to simplify the check on chunk till end of file

Reply:
done

Reply:
After it was done, I took a second look, and have decided to revert it:

The patching logic was spread over multiple places (the ones you proposed + processing's `calculate_unknown_chunks`), and even existing tests had to be modified (`test_search_chunks` + `test_calculate_unknown_chunks`).

One option would be to simplify away patching unknown chunks, but I do not like that.  I think the parent file belongs to the chunk, unknown chunks included.

----

The current patching is also easier to change if we ever want to go with `Path`-s and parallel chunk processing.

Reply:
https://github.com/onekey-sec/unblob/commit/af050d646929d393e6b9c8e1413581e70a8e37eb